common/unsafe/src/main/java/org/apache/spark/unsafe/types/CalendarInterval.java

```diff
@@ -365,12 +368,12 @@ public boolean equals(Object other) {
     if (other == null || !(other instanceof CalendarInterval)) return false;
 
     CalendarInterval o = (CalendarInterval) other;
-    return this.months == o.months && this.microseconds == o.microseconds;
+    return this.months == o.months && this.days == o.days && this.microseconds == o.microseconds;
```

Comment:
Is this line changed manually? We'd better use IDE to generate the equals and hashCode.